### PR TITLE
test: more tests and fix tuple

### PR DESCRIPTION
**Pull request by brisvag**
Added a bunch of tests and improved the ones we have to check more thoroughly for element types. Some of the new tests are supposed to fail, but one of the failing ones is not ideal (the iterable one), though it's an edge case that shouldn't be a big deal.

I also fixed the `Tuple[int, ...]` case, by simply ignoring the ellipsis and falling back to the 1 by 1 case. One downside is that `Tuple[int]` now will accept `(1, 1)`. Need to change that.

### CODE REVIEW

**Comment by brisvag** [comment]
Ok, solved it... it required a bit more noodling, but it's still reasonable.

File: tests/test_pydantic_generics.py [hunk 1]
```diff
@@ -173,21 +217,32 @@ def test_simple_generics(field: type, value: Any) -> None:
...
     (MyList[str], {1}, ['1']),
+    (MyTuple[str], {1}, ('1',)),
     (MyMutableSet[str], [1], {'1'}),
     (MyValidatingMutableSequence[str], {1}, ['1']),
     (MyValidatingList[str], {1}, ['1']),
+    (MyValidatingTuple[str], {1}, ('1',)),
     (MyValidatingMutableSet[str], [1], {'1'}),
+    (MyList[int], '123', [1, 2, 3]),
+    # multiple parameters for iterable will validate one by one
+    (MyTuple[int, float, str], [1, 2, 3], (1, 2.0, '3')),
```
Comment (tlambert03) [inline tests/test_pydantic_generics.py:241]: `MyTuple[int, float, str]` is, itself, a type error, since the generic `MyTuple` type was only defined as taking a single typevar.  (This is part of why it's failing on py3.8 i think)
